Assets/Scripts/Effects/EffectSteps/CheckTypeOfMostRecentlyPlayedCard.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/*
+    This effect step takes two ints, does the boolean comparison on them,
+    and outputs true or false given the operation.
+*/
+public class CheckTypeOfMostRecentlyPlayedCard : EffectStep
+{
+    [SerializeField]
+    private CardCategory cardCategory; 
+    
+    [SerializeField]
+    private string outputKey = "";
+    
+    public CheckTypeOfMostRecentlyPlayedCard() {
+        effectStepName = "CheckTypeOfMostRecentlyPlayedCard";
+    }
+
+    public override IEnumerator invoke(EffectDocument document) {
+        List<Card> cardsCastThisCombat = EnemyEncounterManager.Instance.combatEncounterState.cardsCastThisCombat;
```

Comment:
Looked into .AddRange (what's used to update cardsCast this turn) and verified that this list will in fact preserve ordinality. lit. https://stackoverflow.com/questions/54187471/how-does-listt-addrange-actually-add-the-range-supplied